### PR TITLE
Added instructions on how to configure windows for windows users wanting to use QIIME2.

### PR DESCRIPTION
**Pull request by mcmk3**

### CODE REVIEW

**Comment by mcmk3** [comment]
I have added this section as part of a summer project with the Knight Lab. The majority of the tutorial I have been working on is relevant to Qiita, but this section was considered more appropriate for QIIME 2.
It contains instructions for how to set-up WSL, and install miniconda to allow QIIME 2 installation. It also gives a brief overview of how to configure the WSL to allow GUI programmes to run. While this is not strictly necessary for QIIME 2 it is useful for analysis of data produced with QIIME 2, for example it allows one to use R studio, Spyder, Atom etc from the WSL.

Whether the placement of my section (as a new, seperate section) is correct is unclear, feedback would be welcome.

(Other) Contributers:
@antgonza 
Mehrbod Estaki

**Comment by thermokarst** [comment]
Thanks @mcmk3! I will review this soon (it is currently being considered for the 2020.11 release cycle).

Thanks!

File: source/install/WSL.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    conda env update --file <RELEASE-NAME>.yml
+
+Or install QIIME 2 into a new environment as follows:
+
+.. command-block::
+    :no-exec:
+
+    wget <link from QIIME2 docs>
+    conda env create -n <name of new environment> --file <RELEASENAME>.yml
+    conda activate <name of new environment>
```
Comment (thermokarst) [inline source/install/WSL.rst:43]: Let's drop this section, since it is mostly just duplicating the "Native Install" docs. I think keeping each install guide a modular component will pay off well here.

File: source/install/WSL.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+See also :doc:`native conda installation <native>` for more in-depth instructions. While it is not necessary for QIIME 2, if you wish to use other programmes with graphical interfaces to analyse or view data generated by QIIME 2 (e.g. R studio or Spyder), some further set up is required.
+
+GUI server installation (optional)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (thermokarst) [inline source/install/WSL.rst:48]: What do you think about moving this "GUI server installation" guide to the QIIME 2 Forum, as a "Community Tutorial?"

**Review by thermokarst — CHANGES_REQUESTED** [review]
Thanks so much, @mcmk3! I took a first-pass through this and have provided a few broad comments and questions. Thanks!

File: source/install/WSL.rst (hunk 2 above)
Comment (mcmk3) [reply source/install/WSL.rst:48]: Thank you for the review! Yes, that makes sense, I just included this section because the extra accessibility is very useful, but it is not strictly necessary.

File: source/install/WSL.rst [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+.. command-block::
+    :no-exec:
+    
+    wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+    bash Miniconda3-latest-Linux-x86_64.sh
+    #once installation has completed
+    conda init 
+
+Restart the terminal to have changes take effect. Packages for your project can now be installed in environments, keeping them separate from any other projects you may have, and ensuring that different dependencies do not clash. You can now proceed to :doc:`native conda installation <native>` for instructions on how to install Qiime 2.
```
Comment (thermokarst) [inline source/install/WSL.rst:18]:
```suggestion
```

Oops! Looks like some of this removal was missed in the last round of edits, so re-adding here.

File: source/install/WSL.rst [hunk 4]
```diff
@@ -0,0 +1,25 @@
+Setting up Windows to use QIIME 2
```
Comment (thermokarst) [inline source/install/WSL.rst:1]: To match the move into the "VMs" section, let's rename to match:

```suggestion
Installing QIIME 2 using WSL
```

File: source/install/WSL.rst [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+---------------------------------
+
+This section will demonstrate how to set up windows to allow QIIME 2 installation. Windows now allows users to install linux as a windows subsystem (WSL), the simplest way to use QIIME 2 on a machine with Windows OS is therefore to use this Windows subsystem for Linux.
```
Comment (thermokarst) [inline source/install/WSL.rst:4]: Unfortunately I don't think WSL is available to _many_ Windows users, so maybe we should temper the language used here a little bit.

```suggestion
This tutorial demonstrates how to set up QIIME 2 on the Windows Subsystem for Linux.
```

File: source/install/WSL.rst [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+Please see the `official windows tutorial <https://docs.microsoft.com/en-us/windows/wsl/install-win10>`__ to install WSL2. Then install an appropriate linux distribution e.g. the latest stable version of Ubuntu. Once installation is complete run ``sudo apt-get update`` and ``sudo apt-get upgrade`` to ensure everything is up to date.
```
Comment (thermokarst) [inline source/install/WSL.rst:6]: I suggest removing the Ubuntu-specific instructions here, since Ubuntu is not a prerequisite.

```suggestion
Please see the `official Microsoft guide`_ to install the WSL. Once you have installed and set up the WSL, install a Linux distribution - we suggest the latest stable version of `Ubuntu`_.
```

File: source/install/WSL.rst [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    conda init 
+
+Restart the terminal to have changes take effect. Packages for your project can now be installed in environments, keeping them separate from any other projects you may have, and ensuring that different dependencies do not clash. You can now proceed to :doc:`native conda installation <native>` for instructions on how to install Qiime 2. 
+
+**References**
+
+1. https://docs.microsoft.com/en-us/windows/wsl/install-win10
+
+2. https://wiki.ubuntu.com/WSL
+
```
Comment (thermokarst) [inline source/install/WSL.rst:25]:
```suggestion
.. _official Microsoft guide: https://docs.microsoft.com/en-us/windows/wsl/install-win10
.. _Ubuntu: https://wiki.ubuntu.com/WSL
```

File: source/install/index.rst [hunk 8]
```diff
@@ -9,12 +9,13 @@ scenario.
    :maxdepth: 2
 
    native
+   WSL
```
Comment (thermokarst) [inline source/install/index.rst:12]: Please move this entry into `source/install/virtual/index`.

```suggestion
```

File: source/install/index.rst [hunk 9]
```diff
@@ -9,12 +9,13 @@ scenario.
...
    virtual/index
 
 Recommendations
 ---------------
 
-The :doc:`native conda installation <native>` is generally the recommended
+The :doc: `native conda installation <native>`_ is generally the recommended
```
Comment (thermokarst) [inline source/install/index.rst:18]: Please revert this change (or let me know if you want me to).

File: source/install/index.rst [hunk 10]
```diff
@@ -28,7 +29,7 @@ option for all cases. In general we recommend the following:
 
   - On newer versions of Windows, performing a :doc:`native conda
     installation <native>` in the `Windows Subsystem for Linux`_ usually
-    works well
+    works well. See :doc:`here <WSL>` for instructions on how to set up the Windows Subsystem.
```
Comment (thermokarst) [inline source/install/index.rst:32]:
```suggestion
    works well. See :doc:`the WSL tutorial <virtual/wsl>` for instructions on how to set up the Windows Subsystem.
```

**Review by thermokarst — CHANGES_REQUESTED** [review]
Thanks so much @mcmk3! A few general notes:

- `WSL.rst` should be moved to `source/install/virtual/`, since WSL is another VM method
- Please insert hard line wraps at 80cols (or let me know and I can do it in a squash commit)

This is going to be a great addition, thanks!

**Comment by mcmk3** [comment]
Thank you for all your suggestions! I thing I have implemented all requests now (excepting the hard wrap), its turned out to be quite a small addition in the end, but I hope it is still useful.
If it is simple for you to do a squash commit then this would be great, otherwise I am happy to insert hard line wraps.

**Comment by thermokarst** [comment]
Thanks so much @mcmk3!

> If it is simple for you to do a squash commit then this would be great

I have pushed up the edits, here. Please let me know what you think. If this all looks good to you we can get this merged in time for the 2020.11 release.

>  its turned out to be quite a small addition in the end, but I hope it is still useful.

I think these additions are going to be super useful. Also, I added a link to you Community Contribution tutorial on the forum (via a general search query on the forum, that way other WSL tutorials will show up, too).

**Comment by thermokarst** [comment]
Hey there @mcmk3 - we are getting ready to start the 2020.11 release. I am going to merge this as it is, if you would like to make any adjustments to my edits, let's do that in the 2021.2 dev cycle. Thanks!

**Review by thermokarst — APPROVED** [review]

**Comment by mcmk3** [comment]
@thermokarst It looks great, thank you!
